doc/design/vic-machine/update.md

```diff
@@ -0,0 +1,175 @@
+
+## Show Existing VCH configuration
+Right now, there is no user friendly way to view all configurations of existing VCH. To support update operation, we need to make it easier, otherwise, update only makes things weird.
+
+vic-machine inspect command is available now, just it shows only the execution endpoint of VCH, instead of the configuration. We can extend that command, to support configuration inspection as well.
+
+- vic-machine inspect <--configuration, --conf>
+  show VCH configuration only
+- vic-machine inspect <--execution, -e>
+  show VCH execution endpoint (default behavior)
+
+### Configuration Inspection
+For configuration inspection, two output format should be supported
+
+- yml format
+
+  This should print out all VCH configuration with predefined categories, e.g. general information, storage, network, etc. in user friendly format. Secret information will be hidden or skipped.
+
+Following is sample format:
+
+```
+General:
+	Target: "vc.domain.com"
+	Thumbprint: "98:97:D1:7B:DF:..."
+	OpsUser: "Operator"
+	Name: "VCH-1"
+	Proxies: 
+		- "https://host:port"
+		- "http://host:port"
+Compute:
+	Path: " myCluster/Resources/myRP"
+	CpuLimit: 0 MHz
+	MemoryLimit: 0 MHZ
+Storages:
+	ImageStore: "datastore"
+	VolumeStore:
+		- "ds://datastore/volumes/default": "default"
+		- "nfs://userid@host:port/volumes/": "nfs"
+Networks:
+	BridgeNetwork:
+		PortGroup: "bridge"
+		IpRange: "172.16.0.0/12"
+	ClientNetwork:
+		PortGroup: "client"
+		Gateway: "10.1.0.0/16,10.2.0.0/16:10.0.0.1"
+		...
+	DNSServer:
+		- "8.8.8.8"
+		- "8.8.4.4"
+Certificate:
+	...
+Registries:
+	InsecureRegistries:
+		- "..."
+	RegistryCA:
+		- "ca file"
+Endpoint:
+	Cpu: 1
+	Memory: 2048 MHZ
+	Debug: 3
+	SSHEnabled: true
+```
+
+- Command Option Format
+
+  The yml format output is more readable for VIC admin, but if users want to recreate their vic-machine create or reconfigure command, it's hard to do based on that format, because there is no one on one mapping between the configuration and command options. 
+
+  Here we'd support another option as "vic-machine inspect --cmd --conf", this will print out the vic-machine create command options minus secret information. 
+
+  The sample output:
+```
+--target                vc.domain.com
+--name                  VCH-1
+--compute-resource      /vcqaDC/host/cls/Resources
+--ops-user              operator
+--image-store           datastore1
+--volume-store          ds://datastore/volumes/default:default
+--volume-store          nfs://userid@host:port/volumes/:nfs
+--bridge-network        bridge
+--bridge-network-range  172.16.0.0/12
+...
+```
+
+Based on this output, it's easy to modify existing configuration to reconfigure command options
+
+Note: Special character might be escaped based on current OS platform. But even with this, the command options cannot be copied to other platform.
+
+- Subcommand Inspect (corresponding to reconfigure subcommand option)
```

Comment:
This is used to work with separated update command, cause that will introduce different command options. If that one is dropped, this will not be used as well